docs/rfcs/0004-cirrus-experiments.md

```diff
@@ -0,0 +1,260 @@
+
+Timing: The experimenter id must be set as soon as possible on the application lifecycle, since it is required for all Cirrus requests and should be included in all Glean pings.
+
+#### 2\. Query Nimbus for active features
+
+The /featurelist endpoint is already queried by the application  on initialization and once every hour. Other than these triggers, a feature list request will be made whenever the experiment id is set.
+
+Note The list of active features will be added to each VPN log summary, to aid in debugging branch specific bugs.
+
+#### 3\. Apply features
+
+Feature flags will be used to apply features dynamically. This binds experiments to the Mozilla VPN release cycle, new feature flags cannot be dynamically set on the client  -- even if the experiment treatment is implemented through addons. Experiments may be designed and implemented in Nimbus/Experimenter before a given feature has been released, however users will only actively see treatments being applied once they are on a version of the application that contains the treatment.
+
+Once the feature list is received from Cirrus, it will be parsed and known features will be turned on while unknown features will be discarded.
+
+Since the Cirrus features will be served alongside the list of features on the featurelist endpoint, TaskGetFeatureList will also be reused to apply experiment features. 
+
+Experimental features may have variables associated with them as can be seen by the [Cirrus feature API response](https://github.com/mozilla/experimenter/tree/main/cirrus#output). Each experimental feature will have a settings group associated with it, with the exact same name as the feature. Each variable will have a corresponding setting under the group associated with it, with the exact same name. For example, an experimental feature like so:
+
+```
+{
+
+ "featuresOverwrite":  {  },
+ "experimentalFeatures":  {
+    "onboardingCtaStyleExperiment":  {
+      "color":  "#FF0000",
+      "fontFamily":  "Proxima Nova"
+    },
+  }
+}
+```
+
+Will have a related feature like so:
+
+```
+EXPERIMENTAL_FEATURE(
+      onboardingCtaStyleExperiment, // Feature ID
+      "Experimentations on the onboarding CTA", // Feature name
+      QStringList("color", "fontFamily"),  // Feature variables
+)
+```
+
+The EXPERIMENTAL_FEATURE macro, intentionally doesn't contain the field for declaring if the feature can be flipped on or off. Experimental features must always be "toggleable".
+
+Note Mozilla VPN features may or may not be "toggleable", meaning they may not be allowed to be turned on or off. Experimental features must always be toggleable, otherwise they cannot be applied dynamically. This has consequences for the implementation of the treatment itself. This needs to be considered on a case by case basis.
```

Comment:
```suggestion
> **Note** Mozilla VPN features may or may not be "toggleable", meaning they may not be allowed to be turned on or off. Experimental features must always be toggleable, otherwise they cannot be applied dynamically. This has consequences for the implementation of the treatment itself. This needs to be considered on a case by case basis.
```